var/spack/repos/builtin/packages/hdf/package.py

```diff
@@ -144,6 +144,22 @@ def configure_args(self):
             # We should not specify '--disable-hdf4-xdr' due to a bug in the
             # configure script.
             config_args.append('LIBS=%s' % self.spec['rpc'].libs.link_flags)
+
+        if '+fortran' in self.spec:
+            # https://github.com/Parallel-NetCDF/PnetCDF/issues/61
+            if self.spec.satisfies('%gcc@10:'):
+                flags = {
+                    'FFLAGS': [],
+                    'FCFLAGS': [],
+                }
+                flags['FFLAGS'].append('-fallow-argument-mismatch')
+                flags['FCFLAGS'].append('-fallow-argument-mismatch')
+
+                for key, value in sorted(flags.items()):
+                    if value:
+                        config_args.append('{0}={1}'.format(key, ' '.join(value)))
```

Comment:
Why aren't we using just:
```suggestion
                config_args.extend([
                    'FFLAGS=-fallow-argument-mismatch',
                    'FCFLAGS=-fallow-argument-mismatch',
                }
```
?

Reply:
I was not looking at this critically enough the first time around. I over simplified. I guess I wanted to leave room in case other flags need to be adjusted in the future.  I will reorganize this portion.

Reply:
I am fine with either, is it better to keep it simple and add complexity as needed rather than up front?

Reply:
I'd rather go with the simplest solution, and extend it if need be later.

Reply:
Modified to the simplified solution.